### ISSUE TITLE
Incorrect CA1508 when variable is captured by local function

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#51124]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity2.visualstudio.com/t/Incorrect-CA1508-when-variable-is-captur/1334856)._

---
[severity:It bothers me. A fix would be nice]
The code analysis reports incorrectly that a variable captured by a local function which is not called directly is always `null`.

```cs
namespace Demo
{
    class Test
    {
        static void MaybeCall(System.Action action) => action ();
        public static void F()
        {
            string? x = null;
            void Local() { x = "foo"; }
            Test.MaybeCall (Local);
            if (x is null) // CA1508 'x is null' is always 'true'
            {
                throw new System.InvalidOperationException ();
            }
        }
    }
}
```

---
#### Feedback Bot on 2/9/2021, 06:00 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)
